From ea3edb4813ac96e1881119697cae4756faafe5da Mon Sep 17 00:00:00 2001
From: Sheng Nan <b38800@freescale.com>
Date: Fri, 1 Mar 2013 16:43:29 +0800
Subject: ENGR00252381: mxc_v4l2_capture: ov5640: incorrect return value of
 ov5640_probe

ov5640_probe always return -1 when error happens. Actually we should use the
correct return value. For example, -ENODEV when no ov5640 is found.

Signed-off-by: Sheng Nan <b38800@freescale.com>
---
 drivers/media/video/mxc/capture/ov5640.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/video/mxc/capture/ov5640.c b/drivers/media/video/mxc/capture/ov5640.c
index 6b78e21..ae057c3 100644
--- a/drivers/media/video/mxc/capture/ov5640.c
+++ b/drivers/media/video/mxc/capture/ov5640.c
@@ -1861,7 +1861,7 @@ err2:
 		regulator_put(io_regulator);
 	}
 err1:
-	return -1;
+	return retval;
 }
 
 /*!
-- 
1.7.9.5

